generator_gomod.go

```diff
@@ -54,6 +72,101 @@ func withGomod(g *SourceGenerator, srcSt, worker llb.State, opts ...llb.Constrai
 	}
 }
 
+func (g *SourceGenerator) gitconfigGeneratorScript() llb.State {
```

Comment:
Have you considered utilizing `GIT_ASKPASS` or just more generally git credential helpers?
I think it would probably be worthwhile to handle secrets with a helper of some sort that can just read secrets directly from `/run/secrets` (where buildkit puts them) instead of trying to deal with the configuration file.

Reply:
AFAIK `GIT_ASKPASS` won't work with github sources since username/password authentication is not allowed by github. I'll look into the credential helpers.

Reply:
TIL about `GIT_ASKPASS`

Reply:
done